components/eamxx/src/physics/nudging/atmosphere_nudging.cpp

```diff
@@ -54,50 +54,97 @@ void Nudging::initialize_impl (const RunType /* run_type */)
-  auto qv_h       = Kokkos::create_mirror_view(fields_ext["qv"]);
+  fields_ext_h["qv"]       = Kokkos::create_mirror_view(fields_ext["qv"]);
+  auto qv_h=fields_ext_h["qv"];
   host_views["qv"] = view_1d_host<Real>(qv_h.data(),qv_h.size());
   layouts.emplace("qv", scalar3d_layout_mid);
 
-  fields_ext_3d["horiz_winds"] = view_3d<Real>("horiz_winds",m_num_cols,2,m_num_src_levs);
-  auto horiz_winds_h       = Kokkos::create_mirror_view(fields_ext_3d["horiz_winds"]);
-  host_views["horiz_winds"] = view_1d_host<Real>(horiz_winds_h.data(),horiz_winds_h.size());
-  layouts.emplace("horiz_winds", horiz_wind_layout);
-  
+  fields_ext["u"] = view_2d<Real>("u",m_num_cols,m_num_src_levs);
+  fields_ext_h["u"]       = Kokkos::create_mirror_view(fields_ext["u"]);
+  auto u_h=fields_ext_h["u"];
+  host_views["u"] = view_1d_host<Real>(u_h.data(),u_h.size());
+  layouts.emplace("u", scalar3d_layout_mid);
+
+  fields_ext["v"] = view_2d<Real>("v",m_num_cols,m_num_src_levs);
+  fields_ext_h["v"]       = Kokkos::create_mirror_view(fields_ext["v"]);
+  auto v_h=fields_ext_h["v"];
+  host_views["v"] = view_1d_host<Real>(v_h.data(),v_h.size());
+  layouts.emplace("v", scalar3d_layout_mid);
+
   auto grid_l = m_grid->clone("Point Grid", false);
   grid_l->reset_num_vertical_lev(m_num_src_levs);
   ekat::ParameterList data_in_params;
-  m_fnames = {"T_mid","p_mid","qv","horiz_winds"};
+  //m_fnames = {"T_mid","p_mid","qv","horiz_winds"};
+  m_fnames = {"T_mid","p_mid","qv","u","v"};
   data_in_params.set("Field Names",m_fnames);
   data_in_params.set("Filename",datafile);
   data_in_params.set("Skip_Grid_Checks",true);  // We need to skip grid checks because multiple ranks may want the same column of source data.
-  AtmosphereInput data_input0(m_comm,data_in_params);
-  data_input0.init(grid_l,host_views,layouts);
-  data_input=std::make_shared<AtmosphereInput>(data_input0);
+  data_input = std::make_shared<AtmosphereInput>(m_comm,data_in_params);
+  data_input->init(grid_l,host_views,layouts);
 
   T_mid_ext = fields_ext["T_mid"];
   p_mid_ext = fields_ext["p_mid"];
-  horiz_winds_ext = fields_ext_3d["horiz_winds"];
   qv_ext = fields_ext["qv"];
+  u_ext = fields_ext["u"];
+  v_ext = fields_ext["v"];
   ts0=timestamp();
 
-  //Load first data for before and after
-  data_input->read_variables(0);
+  //Check that internal timestamp starts at same point as time in external file
+  int start_date=scorpio::get_int_attribute_c2f(datafile.c_str(),"start_date");
+  int start_time=scorpio::get_int_attribute_c2f(datafile.c_str(),"start_time");
+  int start_year=int(start_date/10000);
+  int start_month=int((start_date-start_year*10000)/100);
+  int start_day=int(start_date-start_year*10000-start_month*100);
+  int start_hour=int(start_time/10000);
+  int start_min=int((start_time-start_hour*10000)/100);
+  int start_sec=int(start_time-start_hour*10000-start_min*100);
+
+  EKAT_REQUIRE_MSG(start_year==ts0.get_date()[0],
```

Comment:
For clarity (and to avoid relying on internal impl details), instead of `get_date()[k]` and `get_time()[k]`, you can use `get_year()`, `get_month()`, ...

Reply:
fixed